File: Kiss/Project/ProjectType.cs
namespace Kiss.Project
{
    public enum ProjectType
    {
        bin, lib, dyn
    }

    public static class ProjectTypeExtensions
    {
        public static string[] ToList()
        {
            return new string[] { ProjectType.bin.ToString(), ProjectType.lib.ToString(), ProjectType.dyn.ToString() };
        }

        public static ProjectType FromString(string value) => value switch
        {
            "bin" => ProjectType.bin,
            "lib" => ProjectType.lib,
            "dyn" => ProjectType.dyn,
            _ => throw new Exception("Invalid value")
        };
    }


}

File: src/project/project_type.py
from enum import Enum

class ProjectType(str, Enum):
    bin = "bin"
    lib = "lib"
    dyn = "dyn"
    workspace = "workspace"
    
    def __str__(self):
        return self.name
    
    def manifestText(self, projectName: str) -> str:
        return f'@{self.manifestImportText()}("{projectName}")'
        
    def manifestImportText(self) -> str:
        if self is ProjectType.bin:
            return "Bin"
        elif self is ProjectType.lib:
            return "Lib"
        elif self is ProjectType.dyn:
            return "Dyn"
        else:
            raise ValueError(f"Invalid ProjectType: {self}")
        
